hbase-client/src/main/java/org/apache/hadoop/hbase/client/ConnectionRegistryFactory.java

```diff
@@ -18,26 +18,28 @@
 package org.apache.hadoop.hbase.client;
 
 import org.apache.hadoop.conf.Configuration;
-import org.apache.yetus.audience.InterfaceAudience;
 import org.apache.hadoop.hbase.util.ReflectionUtils;
+import org.apache.yetus.audience.InterfaceAudience;
 
 /**
- * Get instance of configured Registry.
+ * Factory class to get the instance of configured connection registry.
  */
 @InterfaceAudience.Private
-final class AsyncRegistryFactory {
+final class ConnectionRegistryFactory {
 
-  static final String REGISTRY_IMPL_CONF_KEY = "hbase.client.registry.impl";
+  static final String CLIENT_CONNECTION_REGISTRY_IMPL_CONF_KEY =
+      "hbase.client.connection.registry.impl";
```

Comment:
I'm not a big fan of the word "registry" either but I think it signifies that it is a one stop shop for all the information needed for connection related metadata (which is probably why it was chosen in the first place). Also, like you said, it has been used in the configs for a while, so...

Reply:
I like this last take of @bharathv  -- don't like it either but does convey one-stop-shop. 

Service is overloaded usually a protobuf Service implemenation such as Admin or Client but also guava Service implementations done internally. If we did Service -- while it might be interesting to expose hbase as a java Service -- I think stuff would get a little confusing when discovery tier talks of Service and then the lower rpc tier talks of another Service type altogether. Locator might work but fails like Service in not being comprehensive enough to cover all exposed in the Interface. Its internal at the moment used in a few places.

Registry while ugly seems the better to me.

Reply:
Bleh. "ClusterInternalServiceDiscoveryService"? I give up and resign myself to "registry".

Reply:
lol :-)